assets/js/base/context/cart-checkout/payment-methods/payment-method-data-context.js

```diff
@@ -341,21 +341,23 @@ export const PaymentMethodDataProvider = ( { children } ) => {
 			return;
 		}
 
-		// If there's no active payment method, or the active payment method has
-		// been removed (e.g. COD vs shipping methods), set one as active.
-		if (
-			! activePaymentMethod ||
-			! paymentMethodKeys.includes( activePaymentMethod )
-		) {
-			setActivePaymentMethod(
-				Object.keys( paymentData.paymentMethods )[ 0 ]
-			);
-		}
+		setActive( ( currentActivePaymentMethod ) => {
+			// If there's no active payment method, or the active payment method has
+			// been removed (e.g. COD vs shipping methods), set one as active.
+			if (
+				! currentActivePaymentMethod ||
+				! paymentMethodKeys.includes( currentActivePaymentMethod )
+			) {
+				dispatch( statusOnly( PRISTINE ) );
+				return Object.keys( paymentData.paymentMethods )[ 0 ];
+			}
+			return currentActivePaymentMethod;
+		} );
 	}, [
-		activePaymentMethod,
 		paymentMethodsInitialized,
 		paymentData.paymentMethods,
-		setActivePaymentMethod,
+		setActive,
+		customerPaymentMethods,
```

Comment:
Is this dependency needed?

Reply:
No, and it looks like it hasn't been needed for a while! Removed it and retested (just in case) 👍